### ISSUE TITLE
🚀 [Feature Request] Any easy way to edit spells?

### ISSUE THREAD

**Issue by patrikpatrik** [cmangos/issues#3287]
### Feature Details

I'm having such a hard time going back and forth with a dbc viewer and making sure the sql queries are right.

But in some cases even changing those don't work. For instance if I want to change my warrior's Charge skill distance to farther than 25 yards but even editing in Spell.dbc and in `spell_template` table in database `tbcmangos` doesnt do anything.

But changing hearthstone cooldown works perfectly fine when doing these two modifications. Using tbcmangos with wow client 2.4.3, windows x64

### Benefits and Purpose

Just a way to make things more easier on spell changes such as cooldown, range distance modifications, and even rage/mana costs.

### Core Versions

- [ ] 1.1.12 (Classic)
- [X] 2.4.3 (The Burning Crusade)
- [ ] 3.3.5a (Wrath of the Lich King)

**Comment by celguar** [comment]
Range checks are also client side so yes your charge change does what it does. But your game won't allow to cast it without client edit

**Comment by patrikpatrik** [comment]
@celguar , does that mean I would need to unpack my mpq files clientside and modify my spell.dbc then repatch it again?

**Comment by celguar** [comment]
Also not sure what you mean by "Editing Spell.dbc". I think the issue is not complexity, is that you need more practice
No editing spells means making a custom .mpq file with a modified .dbc files. Then putting it to Data folder of client

**Comment by celguar** [comment]
1) editing spell_template
2) Modiying Spell.dbc and putting it as a .mpq patch in Data folder

**Comment by celguar** [comment]
There are tools for spell Editing like Stone Harry's spell editor, maybe it can help you. I know it's hard and boring to edit these things

**Comment by patrikpatrik** [comment]
@celguar What I meant by editing Spell.dbc is, in the tbcmangos folder theres a ./dbc folder full of *.dbc after the extraction process. There's a Spell.dbc in there.

For example with hearthstone cooldown, I would need a dbc editor to view Spell.dbc, and change cooldown on Column 25 from 3600000 (which is 1 hour) to lets say 9000 for (9 seconds). Then step 2) go to `tbcmangos` database in mysql, and edit `item_template` with the same value and then restart, clear cache. So far that works.

But you mentioned i would need to edit the client side for range checks, so this means I would need to unpack the mpq files, edit the spell.dbc in there, repatch them back up again for lets say Warrior's charge spell to work?

**Comment by celguar** [comment]
1) Editing Spell.dbc in server files changes nothing, Server uses data from spell_template and other tables
2) Nonetheless you can take that Spell.dbc (or any other dbc) and copy it somwhere, edit something (like range on Charge) and then pack it into a separate .mpq file, call it "patch-X.mpq" or something and put into Data Folder. Note that Structure in that patch should match original. So dbc files should be in some "DBFilesClient" or whatever that folder is in original MPQ's

Unpacking existing archives and changing them is not recommended. There can be several DBC's of same name that are overwritten by next patch data so you may accidentally edit old one that has no effect on the game anymore

So let's repeat:
Editing spells in cmangos
1) editing spell_template (or other tables if needed)
2) Modiying Spell.dbc (or other dbc if needed) and putting it as a .mpq patch in Data folder

**Comment by patrikpatrik** [comment]
@celguar I see. It's very tedious, much more difficult than modifying spells on other emulators. Anyway when I got hearthstone to work I thought it was going to be easy. But I didnt know certain things like range checks were done client side and so forth. Well thanks for the tip. I'll try it out and respond back in time.

**Comment by celguar** [comment]
Other emulators process is same I believe. Unless they use dbc only

**Comment by celguar** [comment]
you can't edit Charge range in ANY emulator without changing client DBC by adding a patch to it anyway
The game uses its own range for Charge and it does not allow you to use it if range is bigger

**Comment by patrikpatrik** [comment]
Sorry, I meant other emulator games like Everquest emulator. I've been messing with that for many many years. But dabbled into mangos until going full into it a week ago. I'm learning! It's very different.

**Comment by celguar** [comment]
Ah, okay. Yes it's all very interesting. Some things you can change server side, of course. Like make intellect buff give more intellect. But a lot of spell data including tooltips is stored in dbc so you won't see new tooltip or new name even if it's changed on the server.

**Comment by patrikpatrik** [comment]
@celguar Very strange. I'm using stone harry's spell editor. I've selected the correct Spell.dbc to examine `Charge` and `Thunder Clap` both Rank 1 and even the other ranks and the ones that had no ranks just to be sure. Modified the range for charge and cooldown for thunder clap. Exported it to patch-4.mpq and placed it in ./data folder and cleared my cache, restarted the server but still not seeing these changes.

I've also updated the mysql database `spell_template` with the correct range values in this case `13` for anywhere and cooldown to 1000 (default 4000 milliseconds). Still everything is at default. What could I be doing wrong?

Is it possible these are just simply hard coded in cmangos in the source code? Sorry for sounding like a complete noob but I've examined the Spell.dbc numerous times to make sure the values were changed, and the tables too.

**Comment by celguar** [comment]
don't forget you have to restart server anytime you change spell related tables

**Comment by celguar** [comment]
also if you don't see numbers like range changing in game, is clearly client patch is not applied properly. either folder structure is not preserved, or something else. Because spell names, range, tooltip, icon client reads from itself

**Comment by patrikpatrik** [comment]
I did. I restarted my server. I made sure I restarted realmd + mangosd. Cleared my cache. double checked, triple checked. Placed patch-4.mpq in the ./data folder in the wow client directory 2.4.3 tbc. What else could I be doing wrong?

**Comment by patrikpatrik** [comment]
I even used myWarCraftStudio to manually inspect the created patch-4.mpq, extracted those dbc files into a temp folder. and re-checked Spell.dbc to make sure the values were updated. I don't need to place patch-4.mpq also inside ./enUS as well do i?

**Comment by patrikpatrik** [comment]
Is it possible there's maybe another table in `tbcmangos` besides `spell_template` that I'm overlooking that needs to be edited too? Unfortunately tables `spell` is empty, `spellcategory`, and `spellrange`.

**Comment by patrikpatrik** [comment]
Ugh.. after countless hours and a sleepless night or two I finally figured it out. So I had to run something called `wowme.exe` which is a patch that fixes this dreaded `ERROR #131 File Corrupt` on `Spell.dbc`. So now I can safely patch mpq and run it and you're right. Range checks are client side and cooldowns I can edit via mysql. Thanks. This matter is now closed I can finally start editing simple spells and trudge forth! 👍
